integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala

```diff
@@ -268,6 +309,151 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
+                  "is not a valid column in table.")
+          }
+        }
+
+        tableProperties.put(CarbonCommonConstants.INDEX_HANDLER +
+          s".$e.sourcecolumntypes", sourceTypes.dropRight(1).toString())
+
+        val handlerType = tableProperties.get(CarbonCommonConstants.INDEX_HANDLER + s".$e.type")
+        val handlerClass = tableProperties.get(CarbonCommonConstants.INDEX_HANDLER + s".$e.class")
+
+        val handlerClassName: String = handlerClass match {
+          case Some(className) =>
+            className
+          case None =>
+            /* use handler type to find the default implementation */
+            if (handlerType.isEmpty) {
+              throw new MalformedCarbonCommandException(
+                s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                  s"Both ${CarbonCommonConstants.INDEX_HANDLER}.$e.class and " +
+                  s"${CarbonCommonConstants.INDEX_HANDLER}.$e.type properties are not specified")
+            } else if (handlerType.get.equalsIgnoreCase("geohash")) {
+              /* Use geoHash default implementation */
+              val className = classOf[org.apache.carbondata.core.util.GeoHashDefault].getName
+              tableProperties.put(s"${CarbonCommonConstants.INDEX_HANDLER}.$e.class", className)
+              className
+            } else {
+              throw new MalformedCarbonCommandException(
+                s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property is invalid. " +
+                  s"Unsupported value : ${handlerType.get} specified in property " +
+                  s"${CarbonCommonConstants.INDEX_HANDLER}.$e.type")
+            }
+        }
+
+        try {
+          val handlerClass: Class[_] = java.lang.Class.forName(handlerClassName)
+          val instance = handlerClass.newInstance().asInstanceOf[
+            CustomIndex[Long, String, java.util.List[Array[Long]]]]
+          instance.validateOption(tableProperties.asJava)
+          instance.init(tableProperties.asJava)
+        } catch {
+          case ex: ClassNotFoundException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"$handlerClassName class in property " +
+                s"${CarbonCommonConstants.INDEX_HANDLER}.$e.class is failed with $ex")
+          case ex@(_: InstantiationError | _: IllegalAccessException) =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Instantiation of class $handlerClassName is failed with $ex")
+          case ex: ClassCastException =>
+            throw new MalformedCarbonCommandException(
+              s"Carbon ${CarbonCommonConstants.INDEX_HANDLER} property process failed. " +
+                s"Failed due to $ex")
```

Comment:
for all the exceptions catched, instead of using due to $ex, better before throwing the MalformedCarbonCommandException, do Logger.error("message", ex) and then can throw our message. Please check for all

Reply:
Modified